bears/general/CasingBear.py

```diff
@@ -0,0 +1,159 @@
+import re
+
+from coalib.bears.LocalBear import LocalBear
+from coalib.bearlib.naming_conventions import (
+    to_camelcase, to_pascalcase, to_snakecase)
+from bears.general.AnnotationBear import AnnotationBear
+from coalib.bearlib.languages.LanguageDefinition import LanguageDefinition
+from coalib.results.SourceRange import SourceRange
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+from coalib.results.HiddenResult import HiddenResult
+
+
+class CasingBear(LocalBear):
+
+    def run(self,
+            filename,
+            file,
+            dependency_results,
+            casing: str,
+            language: str,
+            coalang_dir: str=None):
+        """
+        Checks whether all identifier names (variables, classes, objects)
+        follow a certain naming convention.
+
+        :param filename:           Name of the file that needs to be checked.
+        :param file:               File that needs to be checked in the form
+                                   of a list of strings.
+        :param dependency_results: A dict of dependencies with bear name as
+                                   key and results as value.
+        :param casing:             "camel" for camelCasing or "snake" for
+                                   snake_casing or "pascal" for PascalCasing.
+        :param language:           The language of the file, which is used to
+                                   determine the keywords to ignore.
+        :param coalang_dir:        External directory for the coalang file.
+        """
+        casing_convention = {"camel": to_camelcase,
+                             "pascal": to_pascalcase,
+                             "snake": to_snakecase}
+        if casing not in casing_convention:
+            self.err("Invalid casing convention provided: " + casing)
+            return
+        self.convertor = casing_convention[casing]
+
+        try:
+            coalang = LanguageDefinition(language, coalang_dir=coalang_dir)
+        except:
+            content = ("coalang specification for " + language +
+                       " not found.")
+            yield HiddenResult(self, content)
+            return
+
+        # TODO: Adapt after coala/coala/issues/2200 is solved
+        if "keywords" not in coalang or "special_chars" not in coalang:
+            self.warn("'keywords' and 'special_chars' are necessary fields "
+                      "that are missing from the coalang definition "
+                      "for {}.".format(language))
+            return
+
+        self.keywords = coalang["keywords"]
+        self.delim = re.escape(str(coalang["special_chars"]) + " \t\n")
```

Comment:
I'm not so fond of putting those in class variables, I' m a fan of explicitly passing it around